engine/src/execution_graph/logic_controllers/CacheMachine.cpp

```diff
@@ -474,6 +474,16 @@ std::unique_ptr<ral::frame::BlazingTable> ConcatenatingCacheMachine::pullFromCac
 			tables_holder[i] = std::move(data->decache());
 			table_views[i] = tables_holder[i]->toBlazingTableView();
 		}
+		if( ral::utilities::checkIfConcatenatingStringsWillOverflow(table_views)) {
```

Comment:
The concatenating cache can usually control how much to concatenate, so we want to add this check in the while loop that collects the messages to concatenate. 
For the case of single node processing, we always want to concatenate all. This is set in the creation of the concatenating cache for the final cache:
https://github.com/BlazingDB/blazingsql/blob/branch-0.15/engine/src/CalciteInterpreter.cpp#L98

Usually we would do this by leaving the `cache_settings` in its default setting, where it just reads everything.
https://github.com/BlazingDB/blazingsql/blob/branch-0.15/engine/src/execution_graph/logic_controllers/taskflow/kpair.h#L18

We add a new `cache_setting`: `bool concat_all` that is only for the Concatenating caches, that will force it to read all, regardless of the thresholds. And use that for the creation of the last cache.
https://github.com/BlazingDB/blazingsql/blob/branch-0.15/engine/src/CalciteInterpreter.cpp#L98

Use that new `concat_all` setting to be able to override `checkIfConcatenatingStringsWillOverflow` or the thresholds when collecting all the messages. And if it is going to concat_all and strings will overflow, then it should provide a warning